src/backend/distributed/commands/role.c

```diff
@@ -668,21 +670,42 @@ MakeSetStatementArgument(char *configurationName, char *configurationValue)
 				int intValue;
 				parse_int(configurationValue, &intValue,
 						  (*matchingConfig)->flags, NULL);
-				arg = makeIntConst(intValue, -1);
+				Node *arg = makeIntConst(intValue, -1);
+				args = lappend(args, arg);
 				break;
 			}
 
 			case PGC_REAL:
 			{
-				arg = makeFloatConst(configurationValue, -1);
+				Node *arg = makeFloatConst(configurationValue, -1);
+				args = lappend(args, arg);
 				break;
 			}
 
 			case PGC_BOOL:
 			case PGC_STRING:
```

Comment:
It seems Postgres uses Strings and Identifiers the same. Is there any GUC that is a list of strings and should not be treated as an identifier? How does such a guc behave in Postgres?

Examples of string lists seem to include:
 - `plpgsql.extra_warnings`
 - `plpgsql.extra_errors`
 - `DateStyle`
 - `temp_tablespaces` *
 - `session_preload_libraries` *
 - `shared_preload_libraries` *
 - `local_preload_libraries` *
 - `search_path` *
 - `log_destination`
 - `listen_addresses`
 - `synchronous_standby_names`
 - `wal_consistency_checking`

*Some also have `GUC_LIST_QUOTE` set, which we don't seem to use in our logic. Do you know its significance? Some information seems to be encapsulated here https://github.com/postgres/postgres/blob/c410af098c46949e36607eb13689e697fa2def97/src/backend/utils/adt/ruleutils.c#L2759-L2774

Would be great to have tests for both, quoted and non quoted lists to verify both work.

Reply:
As of latest `REL_12_STABLE` today (https://github.com/postgres/postgres/commit/17d8cf2897443874fb66d2e4895a049cf961ec58) GUCs that have `GUC_LIST_INPUT` is set are all in the following categories:
- The ones that also have `GUC_LIST_QUOTE` flag set. This means that they will not get downcased ever.
  - temp_tablespaces, session_preload_libraries, shared_preload_libraries, local_preload_libraries, search_path
- Enum values where case is not important
  - DateStyle, log_destination, wal_consistency_checking
- Configuration values that are not allowed on `ALTER ROLE .. SET statements.
  - listen_addresses, synchronous_standby_names

I can add some tests for DateStyle changes without breaking tests I guess.